backtracking/subset_sum.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ * @brief Program to Count number of subsets
+ *
+ * @details Subset problem (https://en.wikipedia.org/wiki/Subset_sum_problem)
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+
+/**
+ * @namespace backtracking
+ * @brief subset sum algorithm
+ */
+namespace backtracking {
+/**
+ * @namespace Subsets
+ * @brief Functions for counting subsets in a given array with a given sum
+ * Time Complexity: O(sum*n), where the sum is the ‘target sum’ and ‘n’ is the
```

Comment:
This is not quite true. The complexity is going to be at least exponential (and actually independent on the sum.
```suggestion
 * Time Complexity: O(n * 2^n), where ‘n’ is the number of elements in the given array.
```